core/shared/src/main/scala/cats/effect/internals/IOConnection.scala

```
@@ -139,4 +156,17 @@ private[effect] object IOConnection {
     def tryReactivate(): Boolean =
       state.compareAndSet(null, Nil)
   }
+
+  private final class TrampolinedPair(lh: IOConnection, rh: IOConnection)
+    extends Cancelable.Type with Runnable {
+
+    override def run(): Unit =
+      try lh.cancel()
+      finally rh.cancel()
```

We'll swallow an error here if they both throw. Is that okay?